dockstore-common/src/test/java/io/dockstore/common/yaml/DockstoreYamlTest.java

```diff
@@ -330,4 +331,54 @@ public void testGitReferenceFilter() {
         assertFalse(DockstoreYamlHelper.filterGitReference(Path.of("refs/tags/["), filters));
     }
 
+
+    @Test
+    public void testGetSuggestedDockstoreYamlProperty() {
+        Class dockstoreYamlClass = DockstoreYaml12.class;
+
+        String suggestedProperty = DockstoreYamlHelper.getSuggestedDockstoreYamlProperty(dockstoreYamlClass, "z");
+        assertEquals("Shouldn't suggest a property if there's too many changes that needs to be done", "", suggestedProperty);
+
+        suggestedProperty = DockstoreYamlHelper.getSuggestedDockstoreYamlProperty(dockstoreYamlClass, "workflows");
+        assertEquals("Should return the same property back if it's already a valid property", "workflows", suggestedProperty);
+
+        suggestedProperty = DockstoreYamlHelper.getSuggestedDockstoreYamlProperty(dockstoreYamlClass, "affilliation");
+        assertEquals("affiliation", suggestedProperty);
+
+        suggestedProperty = DockstoreYamlHelper.getSuggestedDockstoreYamlProperty(dockstoreYamlClass, "filter");
+        assertEquals("filters", suggestedProperty);
+
+        suggestedProperty = DockstoreYamlHelper.getSuggestedDockstoreYamlProperty(dockstoreYamlClass, "apptool");
+        assertEquals("tools", suggestedProperty);
+
+        suggestedProperty = DockstoreYamlHelper.getSuggestedDockstoreYamlProperty(dockstoreYamlClass, "e-mail");
+        assertEquals("email", suggestedProperty);
+
+        suggestedProperty = DockstoreYamlHelper.getSuggestedDockstoreYamlProperty(dockstoreYamlClass, "testParameterFilets");
+        assertEquals("testParameterFiles", suggestedProperty);
+    }
+
+    @Test
+    public void testGetDockstoreYamlProperties() {
+        Set<String> properties = DockstoreYamlHelper.getDockstoreYamlProperties(DockstoreYaml12.class);
+        assertEquals("There should be 33 unique properties in a version 1.2 .dockstore.yml", 33, properties.size());
+
+        properties = DockstoreYamlHelper.getDockstoreYamlProperties(DockstoreYaml11.class);
+        assertEquals("There should be 29 unique properties in a version 1.1 .dockstore.yml", 29, properties.size());
+    }
+
+    @Test
+    public void testValidateDockstoreYamlProperties() {
+        try {
+            DockstoreYamlHelper.validateDockstoreYamlProperties(DOCKSTORE12_YAML.replace("publish", "published"));
+        } catch (DockstoreYamlHelper.DockstoreYamlException ex) {
+            assertEquals("Unknown property: 'published'. Did you mean: 'publish'?", ex.getMessage());
```

Comment:
Might want to use a public String constant in `DockstoreYamlHelper` and here to make it obvious that this test needs to be changed if `DockstoreYamlHelper` changes or vice versa. 
e.g. If we change "Unknown property:" to "We found an unknown property"